manifests/init.pp

```diff
@@ -59,20 +62,33 @@
   $manage_epel      = true,
   $install_pkgdeps  = true,
   $install_pip      = true,
+  $provider         = 'pip',
   $proxy            = $awscli::params::proxy,
   $install_options  = $awscli::params::install_options,
 ) inherits awscli::params {
-  class { '::awscli::deps':
-    proxy => $proxy,
-  }
 
-  package { 'awscli':
-    ensure          => $version,
-    provider        => 'pip',
-    install_options => $install_options,
-    require         => [
-      Package[$pkg_pip],
-      Class['awscli::deps'],
-    ],
-  }
+  case $provider {
+    'pip': {
+      class { '::awscli::deps':
+        proxy => $proxy,
+      }
+
+      package { 'awscli':
+        ensure          => $version,
+        provider        => 'pip',
+        install_options => $install_options,
+        require         => [
+          Package[$pkg_pip],
+          Class['awscli::deps'],
+        ],
+      }
+    }
+
+    'apt' : {
+      package { 'awscli':
+        ensure   => 'present',
```

Comment:
Ah, I totally overlooked the fact that it is parameterized. Let's set this to `$version` (which defaults to 'present' but can then be overridden by the user). Apologies!

Reply:
Makes sense – fixed!